### PR TITLE
gramps: 5.1.6 -> 5.2.0, drop legacy builder

### PR DESCRIPTION
**Pull request by TomaSajt**
## Description of changes

This PR updates the `gramps` package to the latest release.

It also drops the old `installPhase` logic, which was a copied-out version of a now removed python builder script.

I changed `pythonPath` over to `propagatedBuildInputs`, not sure what the real differences are.

I could also remove a patch, which was no longer needed.

I added a patch, which disables the incomplete GTK install warning dialog appearing on launch

I fixed testing by using `unittestCheckHook` and creating an empty `.git` directory (so that it know it hasn't been installed yet)

I avoided double-wrapping the executable by using `dontWrapGApps` with `preFixup`

I added the `disabled` attribute using `pythonOlder` 

I added `meta.mainProgram`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by 06kellyjac — APPROVED** [review]
Thanks. Do you want to join the maintainers list?

**Comment by TomaSajt** [comment]
> Thanks. Do you want to join the maintainers list?

Yeah, I could help maintain this package

**Review by niknetniko — APPROVED** [review]
Result of nixpkgs-review:

```
2 packages built:
gramps gramps.dist
```

I poked around a bit, looks like everything works.

**Comment by pinpox** [comment]
nixpkgs-review fails to evaluate for me:

```
│❯ nixpkgs-review pr 291646                                                                                                                                                                                                                           │
│$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/291646/head:refs/nixpkgs-review/1                                                                                              │
│$ git worktree add /home/pinpox/.cache/nixpkgs-review/pr-291646-1/nixpkgs 8687ddab9f7479679b899aa27b63be6334650796                                                                                                                                   │
│Preparing worktree (detached HEAD 8687ddab9f74)                                                                                                                                                                                                      │
│Updating files: 100% (40536/40536), done.                                                                                                                                                                                                            │
│HEAD is now at 8687ddab9f74 Merge pull request #297404 from r-ryantm/auto-update/werf                                                                                                                                                                │
│$ git merge --no-commit --no-ff 4ace37ff4e6b841de9baf28318b39a5d0461267d                                                                                                                                                                             │
│Auto-merging pkgs/applications/misc/gramps/default.nix                                                                                                                                                                                               │
│Automatic merge went well; stopped before committing as requested                                                                                                                                                                                    │
│error:                                                                                                                                                                                                                                               │
│       … while calling the 'listToAttrs' builtin                                                                                                                                                                                                     │
│                                                                                                                                                                                                                                                     │
│         at /nix/store/i47y1c2dcsr877irqrc81qgrlcmhzgip-nixpkgs-review-2.10.3/lib/python3.11/site-packages/nixpkgs_review/nix/evalAttrs.nix:46:1:                                                                                                    │
│                                                                                                                                                                                                                                                     │
│           45|                                                                                                                                                                                                                                       │
│           46| listToAttrs (concatMap getProperties attrs)                                                                                                                                                                                           │
│             | ^                                                                                                                                                                                                                                     │
│           47|                                                                                                                                                                                                                                       │
│                                                                                                                                                                                                                                                     │
│       … while calling the 'concatMap' builtin                                                                                                                                                                                                       │
│                                                                                                                                                                                                                                                     │
│         at /nix/store/i47y1c2dcsr877irqrc81qgrlcmhzgip-nixpkgs-review-2.10.3/lib/python3.11/site-packages/nixpkgs_review/nix/evalAttrs.nix:46:14:                                                                                                   │
│                                                                                                                                                                                                                                                     │
│           45|                                                                                                                                                                                                                                       │
│           46| listToAttrs (concatMap getProperties attrs)                                                                                                                                                                                           │
│             |              ^                                                                                                                                                                                                                        │
│           47|                                                                                                                                                                                                                                       │
│                                                                                                                                                                                                                                                     │
│       (stack trace truncated; use '--show-trace' to show the full trace)                                                                                                                                                                            │
│                                                                                                                                                                                                                                                     │
│       error: attribute 'mainProgram' already defined at /home/pinpox/.cache/nixpkgs-review/pr-291646-1/nixpkgs/pkgs/applications/misc/gramps/default.nix:92:5                                                                                       │
│                                                                                                                                                                                                                                                     │
│       at /home/pinpox/.cache/nixpkgs-review/pr-291646-1/nixpkgs/pkgs/applications/misc/gramps/default.nix:94:5:                                                                                                                                     │
│                                                                                                                                                                                                                                                     │
│           93|     homepage = "https://gramps-project.org";                                                                                                                                                                                          │
│           94|     mainProgram = "gramps";                                                                                                                                                                                                           │
│             |     ^                                                                                                                                                                                                                                 │
│           95|     maintainers = with maintainers; [ jk pinpox tomasajt ];                                                                                                                                                                           │
│https://github.com/NixOS/nixpkgs/pull/291646 failed to build: nix --extra-experimental-features nix-command no-url-literals --system x86_64-linux eval --nix-path nixpkgs=/home/pinpox/.cache/nixpkgs-review/pr-291646-1/nixpkgs nixpkgs-overlays=/tm│
│p/tmp2ll7_bxa --json --impure --no-allow-import-from-derivation --expr (import /nix/store/i47y1c2dcsr877irqrc81qgrlcmhzgip-nixpkgs-review-2.10.3/lib/python3.11/site-packages/nixpkgs_review/nix/evalAttrs.nix { attr-json = /tmp/tmpyz6yu3kl; }) fai│
│led to run, /tmp/tmpyz6yu3kl was stored inspection                                                                                                                                                                                                   │
│$ git worktree remove -f /home/pinpox/.cache/nixpkgs-review/pr-291646-1/nixpkgs 
```

**Comment by TomaSajt** [comment]
It seems that there was a treewide commit, which added `meta.mainProgram`, but it didn't cause a merge conflict somehow. I have rebased to a newer commit

**Comment by pinpox** [comment]
Result of `nixpkgs-review pr 291646` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>gramps</li>
    <li>gramps.dist</li>
  </ul>
</details>

**Comment by pinpox** [comment]
Result of `nixpkgs-review pr 291646` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>gramps</li>
    <li>gramps.dist</li>
  </ul>
</details>

**Comment by pinpox** [comment]
builds fine now, but shows an error when starting up:
image

```
(gramps:61296): Gtk-WARNING **: 09:47:42.842: Locale not supported by C library.
        Using the fallback 'C' locale.
Gdk-Message: 09:47:43.341: Unable to load hand2 from the cursor theme
2024-03-20 09:47:43.343: ERROR: grampsapp.py: line 188: Unhandled exception
Traceback (most recent call last):
  File "/nix/store/znpq183jdi4gfh7mvh2d68pr0gsfv645-gramps-5.2.0/lib/python3.11/site-packages/gramps/gui/grampsgui.py", line 768, in do_activate
    Gramps(self.argparser, self)
  File "/nix/store/znpq183jdi4gfh7mvh2d68pr0gsfv645-gramps-5.2.0/lib/python3.11/site-packages/gramps/gui/grampsgui.py", line 564, in __init__
    from .viewmanager import ViewManager
  File "/nix/store/znpq183jdi4gfh7mvh2d68pr0gsfv645-gramps-5.2.0/lib/python3.11/site-packages/gramps/gui/viewmanager.py", line 76, in <module>
    from .plug import PluginWindows, ReportPluginDialog, ToolPluginDialog
ImportError: cannot import name 'PluginWindows' from 'gramps.gui.plug' (/nix/store/znpq183jdi4gfh7mvh2d68pr0gsfv645-gramps-5.2.0/lib/python3.11/site-packages/gramps/gui/plug/__init__.py)

Gdk-Message: 09:48:12.363: Error flushing display: Broken pipe
```
I don't know much about gramps, let me know if I'm testing it wrong.

**Comment by TomaSajt** [comment]
I am only getting this error on startup 
image

but it works fine after I close the window.

I'll look into where exactly it's failing for you

**Comment by TomaSajt** [comment]
Welp, I don't know what could be going wrong. 
Maybe it's some env-var.
Could you try running it in a clean environment?
```sh
env -i HOME=$HOME DISPLAY=$DISPLAY ./result/bin/gramps
```

**Comment by TomaSajt** [comment]
I pushed a new change, which adds a patch for removing the initial bad GTK install warning screen.

**Review by niknetniko — APPROVED** [review]
Result of nixpkgs-review pr 291646 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

<details>
  <summary>2 packages built:</summary>
  <ul dir="auto">
    <li>gramps</li>
    <li>gramps.dist</li>
  </ul>
</details>

The patch for the incomplete GTK installation also works, thanks a lot for this!

**Comment by TomaSajt** [comment]
Just something of note before merging: the tests are failing on darwin.
I'll disable them on darwin if I get home

**Comment by pinpox** [comment]
> Welp, I don't know what could be going wrong. Maybe it's some env-var. Could you try running it in a clean environment?
> 
> ```shell
> env -i HOME=$HOME DISPLAY=$DISPLAY ./result/bin/gramps
> ```

This fixed the problem for me. Guess it's some impurity in `nixpkgs-review` then

**Review by pinpox — APPROVED** [review]

**Comment by TomaSajt** [comment]
I found [a patch from FreeBSD](https://lists.freebsd.org/archives/dev-commits-ports-all/2024-February/101616.html) which also fixes an issue related to what error is showing on the tests. Hopefully this will fix the checks here as well.

**Comment by TomaSajt** [comment]
Great, it looks like it successfully got build on darwin too!

**Review by 06kellyjac — APPROVED** [review]

**Review by pinpox — APPROVED** [review]

**Comment by 06kellyjac** [comment]
Very nice to see this finally land. Thanks all.

**Pull request merged**
